### PR TITLE
[ARM] `az deployment mg create`: Add new parameter `--mode` to support setting the mode for deploying resources

### PR DESCRIPTION
**Pull request by PARADISSEEKR**
**Related command**

`az deployment mg create` 

**Description**

az deployment mg create --mode
close https://github.com/Azure/azure-cli/issues/24193

**Testing Guide**

**History Notes**

[Component Name 1] BREAKING CHANGE: `az command a`: Make some customer-facing breaking change
[Component Name 2] `az command b`: Add some customer-facing feature

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [ ] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
Resource

File: src/azure-cli/azure/cli/command_modules/resource/_params.py [hunk 1]
```diff
@@ -441,6 +441,7 @@ def load_arguments(self, _):
                    min_api="2019-10-01")
         c.argument('what_if', arg_type=deployment_what_if_type)
         c.argument('proceed_if_no_change', arg_type=deployment_what_if_proceed_if_no_change_type)
+        c.argument('mode', arg_type=get_enum_type(DeploymentMode, default='incremental'), help='The mode that is used to deploy resources. This value can be either Incremental or Complete. In Incremental mode, resources are deployed without deleting existing resources that are not included in the template. In Complete mode, resources are deployed and existing resources in the resource group that are not included in the template are deleted. Be careful when using Complete mode as you may unintentionally delete resources.')
```
Comment (zhoxing-ms) [inline src/azure-cli/azure/cli/command_modules/resource/_params.py:444]: > default='incremental'

Why is the default value of `DeploymentMode` is `incremental`?

File: src/azure-cli/azure/cli/command_modules/resource/custom.py [hunk 2]
```diff
@@ -643,17 +644,19 @@ def validate_arm_template_at_management_group(cmd,
                                                     template_file=template_file, template_uri=template_uri, parameters=parameters,
                                                     deployment_name=deployment_name, deployment_location=deployment_location,
                                                     validate_only=True, no_wait=no_wait,
-                                                    no_prompt=no_prompt, template_spec=template_spec, query_string=query_string)
+                                                    no_prompt=no_prompt, template_spec=template_spec, query_string=query_string,
+                                                    mode='Incremental')
```
Comment (zhoxing-ms) [inline src/azure-cli/azure/cli/command_modules/resource/custom.py:648]: Same as above

File: src/azure-cli/azure/cli/command_modules/resource/_params.py (hunk 1 above)
Comment (PARADISSEEKR) [reply src/azure-cli/azure/cli/command_modules/resource/_params.py:444]: If the `mode` parameter is not passed, the service return mode cannot be `None`, and the original logic is incremental by default.
https://github.com/Azure/azure-cli/blob/4922433eef479449959a5817d237d2326ec30c40/src/azure-cli/azure/cli/command_modules/resource/custom.py#L741-L744

**Comment by zhoxing-ms** [comment]
Could we also add support for `sub` and `tenant` level?

**Comment by PARADISSEEKR** [comment]
> Could we also add support for `sub` and `tenant` level?

`az deployment sub` and `az deployment tenant` can also be supported if necessary, because they use similar logic.

Currently only `az deployment mg create`  needs to be supported in the feature request(https://github.com/Azure/azure-cli/issues/24193),

**Review by zhoxing-ms — APPROVED** [review]

**Pull request merged**